quantinuum_qircheck/qircheck.py

```diff
@@ -161,23 +161,35 @@ def _is_valid_call_help(instr: pq.Instruction) -> bool:
     main_fun = next(filter(pq.is_entry_point, qir_prog.functions), None)
     if not main_fun:
         raise ValueError(
-            "Expected the QIR file to have an entrypoint function " "but none was found"
+            "Expected the QIR file to have an entrypoint function but none was found"
         )
     num_qubits = pq.required_num_qubits(main_fun)
     if not isinstance(num_qubits, int):
         raise ValueError(
-            "Expected the QIR file to have qubit count specified "
-            "but no annotation was found"
+            "Expected the QIR file to have qubit count specified but no annotation was found"
         )
     num_results = pq.required_num_results(main_fun)
     if not isinstance(num_results, int):
         raise ValueError(
-            "Expected the QIR file to have measurement result count "
-            "specified but no annotation was found"
+            "Expected the QIR file to have measurement result count specified but no annotation was found"
         )
 
+    # check for loops in CFG:
```

Comment:
Maybe I'm misunderstanding but the logic seems wrong here. Suppose there are two basic blocks A and B, and A contains an instruction to jump to B, with no jumps the other way. There are no loops in this graph, but if we iterate in the order [B, A] we will raise the error.

Reply:
I was under the assumption that they would always be in topological order as I was not able to construct an example that would violate this.
Nevertheless I have made a suggestion to improve the logic.